vnext/JSI/Shared/ChakraRuntime.cpp

```diff
@@ -607,284 +616,202 @@ facebook::jsi::Runtime::ScopeState *ChakraRuntime::pushScope() {
+      "JsGetExternalData");
   return externalData;
 }
 
-void ChakraRuntime::checkException(JsErrorCode result) {
-  bool hasException = false;
-  if (result == JsNoError && (JsHasException(&hasException), !hasException))
-    return;
-
-  checkException(result, nullptr);
-}
-
-void ChakraRuntime::checkException(JsErrorCode result, const char *message) {
-  bool hasException = false;
-  if (result == JsNoError && (JsHasException(&hasException), !hasException))
-    return;
-
-  std::ostringstream errorStream;
-
-  if (message != nullptr)
-    errorStream << message << ". ";
-
-  if (result != JsNoError) {
-    errorStream << "ChakraCore API Error :" << std::hex << result << ". ";
-  }
-
-  if (hasException || result == JsErrorScriptException) {
-    errorStream << "JS exception found: ";
-
-    JsValueRef exn;
-    JsGetAndClearException(&exn);
-
-    JsPropertyIdRef messageName;
-    JsGetPropertyIdFromName(L"stack", &messageName);
-
-    JsValueRef messageValue;
-    if (JsGetProperty(exn, messageName, &messageValue) == JsNoError) {
-      if (JsConvertValueToString(exn, &messageValue) != JsNoError) {
-        errorStream << "Unable to retrieve JS exception stack";
-      } else {
-        errorStream << JSStringToSTLString(messageValue);
-      }
-    } else {
-      JsValueRef exnJStr;
-      if (JsConvertValueToString(exn, &exnJStr) != JsNoError) {
-        errorStream << "Unable to describe JS exception";
-      } else {
-        errorStream << JSStringToSTLString(exnJStr);
-      }
+void ChakraRuntime::ThrowUponJsError(
+    JsErrorCode error,
+    const char *const chakraApiName) {
+  switch (error) {
+    case JsNoError: {
+      return;
+      break;
     }
-  }
-
-  std::string errorString = errorStream.str();
-  throw facebook::jsi::JSError(
-      *this, createStringFromAscii(errorString.c_str(), errorString.length()));
-}
-
-JsValueRef ChakraRuntime::stringRef(const facebook::jsi::String &str) {
-  return static_cast<const ChakraStringValue *>(getPointerValue(str))->m_str;
-}
-
-JsPropertyIdRef ChakraRuntime::propIdRef(const facebook::jsi::PropNameID &sym) {
-  return static_cast<const ChakraPropertyIdValue *>(getPointerValue(sym))
-      ->m_propId;
-}
-
-JsValueRef ChakraRuntime::objectRef(const facebook::jsi::Object &obj) {
-  return static_cast<const ChakraObjectValue *>(getPointerValue(obj))->m_obj;
-}
-
-JsWeakRef ChakraRuntime::objectRef(const facebook::jsi::WeakObject &obj) {
-  return static_cast<const ChakraWeakRefValue *>(getPointerValue(obj))->m_obj;
-}
-
-facebook::jsi::String ChakraRuntime::createString(JsValueRef str) const {
-  return make<facebook::jsi::String>(makeStringValue(str));
-}
 
-facebook::jsi::PropNameID ChakraRuntime::createPropNameID(JsValueRef str) {
-  return make<facebook::jsi::PropNameID>(makePropertyIdValue(str));
-}
+    case JsErrorScriptException: {
+      JsValueRef jsError;
+      CrashUponChakraError(JsGetAndClearException(&jsError));
```

Comment:
>JsGetAndClearException [](start = 27, length = 22)

There was more code in the old checkException function that tried to retrieve the exception stack / message, or at least convert the thrown value to a string.

Also, it looks like this always terminates the app, which is a functional change. I don't believe we actually want to terminate the app every time there is a script / syntax error in the javascript; we should throw instead and let the current flow take over (red box or whatever).

Reply:
JSError's constructor already extracts stack and message. See [https://github.com/microsoft/react-native/blob/master/ReactCommon/jsi/jsi/jsi.cpp](https://github.com/microsoft/react-native/blob/master/ReactCommon/jsi/jsi/jsi.cpp) at line 375.

I am not sure what you mean by this "always terminates the app." If you are referring to the extra crash I added when we cannot clear the JS exception from the runtime. I can change it so that a JSINativeException is thrown instead.

---
In reply to: [336274511](https://github.com/microsoft/react-native-windows/pull/3445#discussion_r336274511) [](ancestors = 336274511)